### PR TITLE
Remove all any's and modernise package

### PR DESCRIPTION
**Pull request by PodaruDragos**
## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by PodaruDragos** [comment]
@notaphplover can you please take a look over this ? 
Also maybe you know how to type the last any there ( i left a TODO: comment)

File: src/server.ts [hunk 1]
```diff
@@ -241,23 +244,24 @@ export class InversifyExpressServer {
...
-
                 // invoke controller's action
+
+                // TODO: Cast this some how ??
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
                 const value = await httpContext.container.getNamed<any>(
                     TYPE.Controller,
                     controllerName,
-                )[key](...args);
+                )?.[key](...args);
```
Comment (notaphplover) [inline src/server.ts:264]: Regarding this `any`, just add some typings. I'm adding a code sample, I do not exactly know which are the right types here, but my approach is the following one:

```ts
type ControllerHandler = (...params: Array<unknown>) => unknown;
type Controller = Record<string, ControllerHandler>;

const service: Controller = httpContext.container.getNamed<Controller>(
    TYPE.Controller,
    controllerName,
);

const controllerHandler: ControllerHandler = service[key] as ControllerHandler;

const value: unknown = await controllerHandler(...args);
```

keep in mind `ControllerHandler` and `Controller` may not be the right types

**Comment by PodaruDragos** [comment]
@dcavanagh Can you please take a look at this ?
if you are ok with this please merge it in master so i can take care of the rest remaining PR's here.

**Comment by LeonardSSH** [comment]
LGTM!

**Review by LeonardSSH — APPROVED** [review]

**Pull request merged**